lib/Plack/Middleware/Recorder.pm

```diff
@@ -22,13 +22,26 @@ sub prepare_app {
     $self->start_url('/recorder/start') unless defined $self->start_url;
     $self->stop_url('/recorder/stop')   unless defined $self->stop_url;
 
-    my $output = $self->{'output'};
+    my $output = delete $self->{output};
     croak "output parameter required" unless defined $output;
 
-    $output = $self->{'output'} = IO::File->new($output, 'w') or croak $!
-        unless ref $output;
+    if (ref $output) {
```

Comment:
There's no way to tell what the value for the ```psgi.run_once``` flag is during prepare_app().  It's not until we're handling the first request that we get passed the PSGI environment and can look at the flag.